processing/src/main/java/io/druid/segment/IndexMergerV9.java

```
@@ -536,73 +541,78 @@ private void makeInvertedIndexes(
-          bitmapToWrite = nullRowBitmap.union(bitmapToWrite);
-        }
-        bitmapIndexWriters.get(dimIndex).write(bitmapToWrite);
-
-        if (spatialIndexWriter != null) {
-          String dimVal = dimVals.get(dictId);
-          if (dimVal != null) {
-            List<String> stringCoords = Lists.newArrayList(SPLITTER.split(dimVal));
-            float[] coords = new float[stringCoords.size()];
-            for (int j = 0; j < coords.length; j++) {
-              coords[j] = Float.valueOf(stringCoords.get(j));
+          ImmutableBitmap bitmapToWrite = bitmapSerdeFactory.getBitmapFactory().makeImmutableBitmap(bitset);
+          if ((dictId == 0) && (Iterables.getFirst(dimVals, "") == null)) {
+            bitmapToWrite = nullRowBitmap.union(bitmapToWrite);
+          }
+          bitmapIndexWriters.get(dimIndex).write(bitmapToWrite);
+
+          if (spatialIndexWriter != null) {
+            String dimVal = dimVals.get(dictId);
+            if (dimVal != null) {
+              List<String> stringCoords = Lists.newArrayList(SPLITTER.split(dimVal));
+              float[] coords = new float[stringCoords.size()];
+              for (int j = 0; j < coords.length; j++) {
+                coords[j] = Float.valueOf(stringCoords.get(j));
+              }
+              tree.insert(coords, bitset);
             }
-            tree.insert(coords, bitset);
           }
         }
+        if (spatialIndexWriter != null) {
+          spatialIndexWriter.write(ImmutableRTree.newImmutableFromMutable(tree));
+        }
+        log.info(
+            "Completed dim[%s] inverted with cardinality[%,d] in %,d millis.",
+            dimension,
+            dimVals.size(),
+            System.currentTimeMillis() - dimStartTime
+        );
       }
-      if (spatialIndexWriter != null) {
-        spatialIndexWriter.write(ImmutableRTree.newImmutableFromMutable(tree));
+      finally {
+        ByteBufferUtils.unmap(dimValsMapped);
```

@leventov, I think @drcrallen is concerned that if the try block throws an exception, and then ByteBufferUtils.unmap also throws an exception, that would mask the original exception (see https://github.com/google/guava/wiki/ClosingResourcesExplained).